detect_secrets/plugins/common/constants.py

```diff
@@ -4,7 +4,7 @@
 WHITELIST_REGEXES = [
     re.compile(r)
     for r in [
-        r'[ \t]+{} *pragma: ?whitelist[ -]secret{}[ \t]*$'.format(start, end)
+        r'[ \t]+{} *pragma: ?whitelist[ -]secret.*{}[ \t]*$'.format(start, end)
```

Comment:
For performance reasons, we can make this regex [lazy](https://stackoverflow.com/questions/2301285/what-do-lazy-and-greedy-mean-in-the-context-of-regular-expressions), e.g. `.*?` in this case, (I think.)

Reply:
Yep! It's better to search for the shortest match. Fixing it now.